hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/FilePerBlockStrategy.java

```diff
@@ -307,11 +304,14 @@ public FileChannel getChannel(File file, boolean sync)
       }
     }
 
-    private static OpenFile open(File file, boolean sync) {
+    private static OpenFile open(Container container, BlockID blockID, boolean sync) {
       try {
+        File file = getChunkFile(container, blockID);
         return new OpenFile(file, sync);
       } catch (FileNotFoundException e) {
         throw new UncheckedIOException(e);
+      } catch (StorageContainerException e) {
+        throw new UncheckedIOException(e);
       }
     }
 
```

Comment:
The close & isOpen function in line 318 & 324 respectively need to have the argument of class ContainerBlockId and not File since the key of the cache has changed.